infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/environment/convert/DockerEnvironmentConverter.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.docker.environment.convert;
+
+import org.eclipse.che.api.core.ValidationException;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalEnvironment;
+import org.eclipse.che.workspace.infrastructure.docker.model.DockerEnvironment;
+
+/**
+ * Converts {@link InternalEnvironment} into {@link DockerEnvironment}.
+ *
+ * @author Alexander Andrienko
+ */
+public interface DockerEnvironmentConverter {
+
+  /**
+   * Returns {@link DockerEnvironment} that is converted from the specified {@link
+   * InternalEnvironment}.
+   *
+   * @param environment environment to converting
+   */
```

Comment:
Why we need that?
Make the object you need for runtime inside InternalEnvironment with factory.

Reply:
The same topic is discussed under the following [comment]( https://github.com/eclipse/che/pull/7126#discussion_r151604958)